File: frontend/src/typings/api/get-individual-leaderboard-response.ts
/* tslint:disable */
import { UserLeaderboardEntryDto } from './user-leaderboard-entry-dto';

export interface GetIndividualLeaderboardResponse {
  leaderboard: Array<UserLeaderboardEntryDto>;
  page_count: number;
  user_ranking: UserLeaderboardEntryDto;
}

File: backend/crates/api/src/models/http/responses/get_individual_leaderboard_response.rs
use crate::models::user::UserLeaderboardEntryDto;

#[derive(serde::Serialize, serde::Deserialize, rocket_okapi::JsonSchema)]
pub struct GetIndividualLeaderboardResponse {
    pub leaderboard: Vec<UserLeaderboardEntryDto>,
    pub page_count: usize,
    pub user_ranking: UserLeaderboardEntryDto,
}
